code/synchronization/forward-trigger-service.py

```diff
@@ -127,25 +119,43 @@ async def forward_signals(serial_port: str, async_q: janus.AsyncQueue[int]) -> N
 def _trigger(sync_q: janus.SyncQueue[int]) -> None:
     """
     Trigger the signal sync queue.
-
     Put a signal into the sync queue.
-
     Parameters:
         sync_q (janus.SyncQueue[int]): The sync queue for triggering the signal.
-
     """
     sync_q.put(b"\x01")
     sync_q.join()
     logging.info("Scanner trigger received")
 
 
+def check_usb_device_connected(usb_vendor_id, usb_product_id):
```

Comment:
Since this function raises exception if not connected, perhaps it would be better to call it differently (this is just a suggestion):

```suggestion
def ensure_usb_device_connected(usb_vendor_id, usb_product_id):
```